x/gov/keeper/msg_server.go

```diff
@@ -85,9 +87,19 @@ func (k msgServer) SubmitProposal(goCtx context.Context, msg *v1.MsgSubmitPropos
 	}
 
 	proposalType := msg.ProposalType
-	if msg.Expedited { // checking for backward compatibility
-		proposalType = v1.ProposalType_PROPOSAL_TYPE_EXPEDITED
+	switch proposalType {
```

Comment:
I think I should move those in the keeper and leave the message server be proposal types agnostic.